UnityProject/Assets/Scripts/Player/JoinedViewer.cs

```diff
@@ -43,29 +43,31 @@ public override void OnStartLocalPlayer()
 			{
 				CmdServerSetupPlayer(GetNetworkInfo(),
 					PlayerManager.CurrentCharacterSettings.Username, DatabaseAPI.ServerData.UserID, GameData.BuildNumber,
-					DatabaseAPI.ServerData.IdToken);
-				CmdServerRequestLoadedScenes(SceneManager.GetActiveScene().name);
+					DatabaseAPI.ServerData.IdToken, SceneManager.GetActiveScene().name);
+
 			}
 		}
 
+
+
 		private async void HandleServerConnection()
 		{
 			await ServerSetUpPlayer(GetNetworkInfo(),
 				PlayerManager.CurrentCharacterSettings.Username, DatabaseAPI.ServerData.UserID, GameData.BuildNumber,
-				DatabaseAPI.ServerData.IdToken);
+				DatabaseAPI.ServerData.IdToken, "");
 			ClientFinishLoading();
 		}
 
 		[Command]
 		private void CmdServerSetupPlayer(string unverifiedClientId, string unverifiedUsername,
-			string unverifiedUserid, int unverifiedClientVersion, string unverifiedToken)
+			string unverifiedUserid, int unverifiedClientVersion, string unverifiedToken, string clientCurrentScene)
 		{
 			ClearCache();
-			_ = ServerSetUpPlayer(unverifiedClientId, unverifiedUsername, unverifiedUserid, unverifiedClientVersion, unverifiedToken);
+			_ = ServerSetUpPlayer(unverifiedClientId, unverifiedUsername, unverifiedUserid, unverifiedClientVersion, unverifiedToken,clientCurrentScene);
 		}
 
-		[Command]
-		private void CmdServerRequestLoadedScenes(string AlreadyLoaded)
+
+		private void ServerRequestLoadedScenes(string AlreadyLoaded)
 		{
```

Comment:
so this is server only now?

Reply:
should be triggered by ServerSetUpPlayer now

Reply:
should it have the ``[Server]`` decorator?

Reply:
ok